code401Challenges/src/main/java/code401Challenges/hashtable/Hashtable.java

```diff
@@ -0,0 +1,79 @@
+package code401Challenges.hashtable;
+
+
+
+public class Hashtable<V> {
+
+    // Instance variables
+    // Set to generic for different possible key:value pairs
+
+    private Node<String, V> [] map;
+    private final int initialSize = 1024;
+    private final int primeValue = 4096;
+
+    //Constructor
+
+    public Hashtable() {
+
+        this.map = new Node[initialSize];
+    }
+
+    // Methods
+
+    //Hash the key
+    public int hash(String key){
+        int hashValue = 0;
+
+        for (int i = 0; i < key.length(); i++){
+            hashValue += ( (int) key.charAt(i) + primeValue) * 3;
+        }
+
+        return hashValue % map.length;
+    }
+
+    // Add the key value to the hash table
+    public void add(String key, V value){
```

Comment:
You don't check if the key is already in the table. Each key needs to be unique! Keys can have multiple values attached, but you need a way to handle that, with this implementation the first key added is the only one returned, and it can't be overwritten